storage/pruning/pruningStorer.go

```diff
@@ -807,6 +766,10 @@ func (ps *PruningStorer) extendSavedEpochsIfNeeded(header data.HeaderHandler) bo
 		return true
 	}
 
+	return ps.persisterNeeded()
+}
+
+func (ps *PruningStorer) persisterNeeded() bool {
```

Comment:
also here. This is not properly overloaded.

Reply:
Refactored.